### DISCUSSION TITLE
Best Practices around the License Key

### DISCUSSION THREAD

**Discussion by dmh143** [mui/mui-x#2323] [Archive]
I hope this is an OK place to ask, just looking for some advice on how all of you are using the License Key without exposing it on your sites.

I am working with an organization where the current architecture is a static front-end site and Apollo/GraphQL for the data - no node server to an API etc. They want to use MUI X for their tables but we have some tables that are not behind authentication.

Tables behind auth are a non-issue, just curious what the rest of you are doing when you are not behind auth. If I inject the key via environment vars the key can be snooped via dev tools in the bundle. Are people using SSR to keep the call to the `setLicenseKey` function hidden or am I even overthinking this?

Apologies if this is not the right place, thank you 🙏

**Comment by kameshraj** [comment]
trying to find best practises for handling license keys. Any recommendations?

**Comment by oliviertassinari** [comment]
> some advice on how all of you are using the License Key without exposing it on your sites.

We have retired GitHub discussions. However, I think https://mui.com/x/introduction/licensing/#license-key-security can answer the question.
